### ISSUE TITLE
Different behavior after migrating the checkpoint from Version 1 to Version 2

### ISSUE THREAD

**Issue by atebbifakhr** [OpenNMT/OpenNMT-tf#620]
Hi,

I've migrated my previous models trained with version 1.15, but I'm not getting the same translation output while the input text and all the related parameters like beam_width and length_penalty. 

Am I missing something or this is something expected?

**Comment by guillaumekln** [comment]
Hi,

At least for Transformer models, we should expect the same translation output. So here are some questions:

* What model type is it?
* By version 1.15, you mean TensorFlow 1.15, right?
* Can you give some examples of the differences you observed?

**Comment by atebbifakhr** [comment]
Yes, the model type is Transformer.
No, I mean, I trained my model with OpenNMT-tf 1.15 before, Tensorflow version was 1.11.

For instance it's the difference for a single sentence:

Yesterday , the ladies of and I was informed of the succulent dishes of **Mrs** La Güera , and I can tell you that they are **guarantees** .

Yesterday , the ladies of and I was informed of the succulent dishes of La Güera , and I can tell you that they are **guaranteed** .

There is a slight difference.

**Comment by guillaumekln** [comment]
For the V1 model, did you run the translation with 1.15 or with the latest 1.25 release?

**Comment by atebbifakhr** [comment]
no, with 1.15 OpenNMT and 1.11 Tensorflow, and witth 2.0 OpenNMT and 2.0 Tensorflow.

**Comment by guillaumekln** [comment]
Ok, you should try running the translation with 1.25 and see if there are differences as well. 1.15 is quite old and it is possible we made some small decoding changes since then.

**Comment by atebbifakhr** [comment]
Thanks, I guess it is related to dropout.

Is it possible to enable the dropout during the inference?

**Comment by guillaumekln** [comment]
You would need to change the code. A minimal change would be to ignore the training flag in the [dropout function](https://github.com/OpenNMT/OpenNMT-tf/blob/master/opennmt/layers/common.py#L9).

**Comment by atebbifakhr** [comment]
Thanks a lot!
